oauth2_http/java/com/google/auth/oauth2/GoogleCredentials.java

```diff
@@ -205,6 +249,20 @@ public GoogleCredentials createWithQuotaProject(String quotaProject) {
     return this.toBuilder().setQuotaProjectId(quotaProject).build();
   }
 
+  /**
+   * Returns the universe domain for the credential
+   *
+   * @return An explicit universe domain if it was explicitly provided, {@code
+   *     GoogleAuthUtils.GOOGLE_DEFAULT_UNIVERSE} otherwise.
+   */
+  public String getUniverseDomain() {
+    if (this.universeDomain == null || universeDomain.trim().isEmpty()) {
```

Comment:
Why not just set universeDomain to the default value if none is provided, and then we can avoid this altogether and just return it directly?

Reply:
To implement the alternative you will have to implement checks and test them in all creds now and in the future. Plus take into account edge cases like deserealization of the old credential versions...